vmm_core/virt_mshv/src/lib.rs

```diff
@@ -1141,22 +1168,23 @@ impl virt::PartitionMemoryMap for MshvPartitionInner {
         }
         let slot_to_use = slot_to_use.unwrap();
 
-        let mut flags = HV_MAP_GPA_READABLE;
+        let mut flags = 0;
         if writable {
-            flags |= HV_MAP_GPA_WRITABLE;
+            flags |= set_bits!(u8, MSHV_SET_MEM_BIT_WRITABLE);
         }
         if exec {
-            flags |= HV_MAP_GPA_EXECUTABLE;
+            flags |= set_bits!(u8, MSHV_SET_MEM_BIT_EXECUTABLE);
         }
 
         let mem_region = mshv_user_mem_region {
             size: size as u64,
             guest_pfn: addr,
             userspace_addr: data as u64,
-            flags,
+            flags: flags,
+            rsvd: [0; 7],
         };
 
-        self.vmfd.map_user_memory(mem_region)?;
+        self.vmfd.map_user_memory(mem_region).expect("cannot map user memory");
```

Comment:
Is it OK to use '.expect("cannot map user memory");' here? The error was previously silent, and failing when trying to unmap the memory. The intent here is to fail closer to the location where the error actually happened.

Reply:
That's weird that the error was silent. Where was it getting lost?

Reply:
Let me find it out and report back.

Reply:
Reverted in the latest iteration